src/main/groovy/io/codearte/gradle/nexus/infra/SimplifiedHttpJsonRestClient.groovy

```diff
@@ -20,58 +23,59 @@ class SimplifiedHttpJsonRestClient {
         GET, POST
     }
 
-    private final RESTClient restClient
-    private final String username
-    private final String password
+    private final OkHttpClient restClient
 
-    SimplifiedHttpJsonRestClient(RESTClient restClient, String username, String password) {
-        this.restClient = restClient
-        this.username = username
-        this.password = password
-//        params.requestContentType = ContentType.JSON  //Does not set Content-Type header as required by WireMock
-        restClient.headers["Content-Type"] = "application/json"   //reported as error in Idea (effectively works fine)
-//        restClient.headers.put("Content-Type", "application/json")
+    SimplifiedHttpJsonRestClient(OkHttpClient restClient, String username, String password) {
+        OkHttpClient.Builder clientBuilder = restClient.newBuilder()
+        clientBuilder.addInterceptor(new Interceptor() {
+            @Override
+            Response intercept(@NotNull Interceptor.Chain chain) throws IOException {
+                Request.Builder request = chain.request().newBuilder()
+                    .header("Content-Type", "application/json")
+                    .header("Accept", "application/json")
```

Comment:
Is there any ContentType-like class in OkHttp  that could be used to get that "application/json" string?

Reply:
Unfortunately seems to only support parsing with no list of constants

https://square.github.io/okhttp/4.x/okhttp/okhttp3/-media-type/#companion-object-functions

Reply:
Ok, in that case maybe just a constant String in that class?